src/cls/async-hooks.ts

```diff
@@ -55,17 +55,19 @@ export class AsyncHooksCLS<Context extends {}> implements CLS<Context> {
     // Create the hook.
     this.hook = this.ah.createHook({
       init: (id: number, type: string, triggerId: number, resource: {}) => {
+        const currentId = this.ah.executionAsyncId();
```

Comment:
```suggestion
        // init is called when a new AsyncResource is created. We want code
        // that runs within the scope of this new AsyncResource to see the same
        // context as its "parent" AsyncResource.
        const currentId = this.ah.executionAsyncId();
        // If the parent doesn't have an associated context, skip.
        if (this.context[currentId] !== undefined) {
          return;
        }
        // The criteria for parent depends on the type of AsyncResource.
        if (type === 'PROMISE') {
          // zig.
        } else {
          // zag.
        }
```

Reply:
I realize that having an undefined current execution context is problematic, but the suggestion as-is doesn't allow trigger ID to be respected when current execution context is undefined, which I think might prevent legitimate use cases of the trigger ID to be ignored. If it's for the sake of cleaner code, PTAL at the change I made instead.